sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -348,7 +348,7 @@ static boolean printHelpUsageAndExitIfNeeded(ListMultimap<String, String> option
             getRegisteredOptions(),
             new Predicate<Class<? extends PipelineOptions>>() {
               @Override
-              public boolean apply(Class<? extends PipelineOptions> input) {
+              public boolean apply(@Nonnull Class<? extends PipelineOptions> input) {
```

Comment:
Got it. That's what I thought. I looked only at the overview list of methods in [the javadoc](https://google.github.io/guava/releases/19.0/api/docs/com/google/common/base/Predicate.html), where such annotations are neglected. If I had scrolled down I would have seen it. What you have is definitely best.